x-pack/legacy/plugins/uptime/public/components/functional/alerts/alert_monitor_status.tsx

```diff
@@ -0,0 +1,334 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useState, useEffect } from 'react';
+import {
+  EuiExpression,
+  EuiFieldNumber,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiPopover,
+  EuiSelectable,
+  EuiSpacer,
+  EuiSwitch,
+  EuiTitle,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { KueryBar } from '../../connected/kuerybar/kuery_bar_container';
+
+interface AlertNumberFieldProps {
+  'data-test-subj': string;
+  disabled: boolean;
+  fieldValue: number;
+  setFieldValue: React.Dispatch<React.SetStateAction<number>>;
+}
+
+export const AlertFieldNumber = ({
+  'data-test-subj': dataTestSubj,
+  disabled,
+  fieldValue,
+  setFieldValue,
+}: AlertNumberFieldProps) => {
+  const [isInvalid, setIsInvalid] = useState<boolean>(false);
+
+  return (
+    <EuiFieldNumber
+      compressed
+      data-test-subj={dataTestSubj}
+      min={1}
+      onChange={e => {
+        const num = parseInt(e.target.value, 10);
+        if (isNaN(num) || num < 1) {
+          setIsInvalid(true);
+        } else {
+          if (isInvalid) setIsInvalid(false);
+          setFieldValue(num);
+        }
+      }}
```

Comment:
not a big deal, but i think generally writing incline function isn't better for code. readability

Reply:
++ - I wanted to extract this function out but didn't want to deal with providing out-of-scope variables to it at the time. I can remove it from the component (will make it easier to test as well) and add params for higher scope vars.

Reply:
Extracted and tested: https://github.com/elastic/kibana/pull/57919/commits/62ed1c6494e8bb79f7757b894ff9067ec5314bd1